pkg/consensus/consensus.go

```diff
@@ -281,7 +281,6 @@ func (s *service) Start() {
 		b, _ := s.Chain.GetBlock(s.Chain.CurrentBlockHash()) // Can't fail, we have some current block!
 		s.lastTimestamp = b.Timestamp
 		s.dbft.Start(s.lastTimestamp * nsInMs)
-		s.Chain.SubscribeForBlocks(s.blockEvents)
 		go s.eventLoop()
```

Comment:
@roman-khimov, do you think it's legit to move block subscription to `s.eventLoop`? I suspect this commit is responsible for recently discovered test failures, because all of them use suspended CLI executor and run chain manually somewhere in the end of the tests.

Initially I requested this change because proper subscription requires blockchain to be running (`notificationDispatcher` loop should be running) and sometimes we may have services started before the blockchain start. So this change is needed for proper services start.

Reply:
It doesn't really matter in normal case, nothing but event loop needs these events. And nothing else affects dbft state. But the old code looked a bit more consistent to me.

At the same time, if you're doing something special in tests you might as well pass some wrapped chain into consensus (it only needs a `Ledger` interface, remember?) with subscription method overridden in some way.

Reply:
We can't preserve the previous code because it may lead to a deadlock (not in tests, but in the node).